state/snapshotsManager_test.go

```diff
@@ -0,0 +1,452 @@
+)
+
+func getDefaultSnapshotManagerArgs() state.ArgsNewSnapshotsManager {
+	return state.ArgsNewSnapshotsManager{
+		ShouldSerializeSnapshots: false,
+		ProcessingMode:           common.Normal,
+		Marshaller:               &marshallerMock.MarshalizerMock{},
+		AddressConverter:         &testscommon.PubkeyConverterMock{},
+		ProcessStatusHandler:     &testscommon.ProcessStatusHandlerStub{},
+		StateMetrics:             &stateTest.StateMetricsStub{},
+		AccountFactory:           &stateTest.AccountsFactoryStub{},
+		ChannelsProvider:         iteratorChannelsProvider.NewUserStateIteratorChannelsProvider(),
+	}
+}
+
+func TestNewSnapshotsManager(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil marshaller", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.Marshaller = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilMarshalizer, err)
+	})
+
+	t.Run("nil process status handler", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.ProcessStatusHandler = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilProcessStatusHandler, err)
+	})
+
+	t.Run("nil address converter", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.AddressConverter = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilAddressConverter, err)
+	})
+
+	t.Run("nil state metrics", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.StateMetrics = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilStateMetrics, err)
+	})
+
+	t.Run("nil channels provider", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.ChannelsProvider = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilChannelsProvider, err)
+	})
+
+	t.Run("nil account factory", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		args.AccountFactory = nil
+
+		sm, err := state.NewSnapshotsManager(args)
+		assert.Nil(t, sm)
+		assert.Equal(t, state.ErrNilAccountFactory, err)
+	})
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		args := getDefaultSnapshotManagerArgs()
+		sm, err := state.NewSnapshotsManager(args)
+		assert.NotNil(t, sm)
+		assert.NoError(t, err)
+	})
+}
+
+func TestSnapshotsManager_SetSyncer(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil syncer", func(t *testing.T) {
+		t.Parallel()
+
+		sm, _ := state.NewSnapshotsManager(getDefaultSnapshotManagerArgs())
+		assert.Equal(t, state.ErrNilTrieSyncer, sm.SetSyncer(nil))
+	})
+
+	t.Run("set syncer concurrently", func(t *testing.T) {
+		t.Parallel()
+
+		sm, _ := state.NewSnapshotsManager(getDefaultSnapshotManagerArgs())
+
+		numConcurrentCalls := 10
```

Comment:
invalid test. Please add wait group operation here. Might add a `time.Sleep(10 * time.Millisecond)` on L129, before the SetSyncer call.